pkg/filesystem/subvolume.go

```diff
@@ -131,6 +139,36 @@ func getK8sRefSubvolume(ctx context.Context, clientsets *k8sutil.Clientsets) map
 	return subvolumeNames
 }
 
+// getk8sRefSnapshotHandle returns the snapshothandle for k8s ref of the volume snapshots
+func getK8sRefSnapshotHandle(ctx context.Context, clientsets *k8sutil.Clientsets) map[string]snapshotInfo {
+
+	Snap, err := snapclient.NewForConfig(clientsets.KubeConfig)
+	if err != nil {
+		logging.Fatal(err)
+	}
+	snapList, err := Snap.VolumeSnapshotContents().List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching volumesnapshotcontents: %v\n", err))
+	}
+
+	snapshotHandles := make(map[string]snapshotInfo)
+	for _, snap := range snapList.Items {
+		if snap.Status != nil && snap.Status.SnapshotHandle != nil {
+			// get the snaps id from snapshot handle
+			splitSnapshotHandle := strings.SplitAfterN(*snap.Status.SnapshotHandle, "-", 6)
```

Comment:
Could you add a unit test for this string parsing? It will help show the need, as well as prevent regressions in the future. If it's simpler for unit testing, you might want to refactor this parsing into a separate helper method.